File: shared/src/main/java/io/github/robertomike/hefesto/exceptions/HefestoException.java
package io.github.robertomike.hefesto.exceptions;


/**
 * This class is for the exceptions that can be thrown
 */
public class HefestoException extends RuntimeException {
    /**
     * constructor
     * @param message the message
     */
    public HefestoException(String message) {
        super(message);
    }

    /**
     * constructor
     * @param message the message
     * @param cause the cause
     */
    public HefestoException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: shared/src/main/kotlin/io/github/robertomike/hefesto/exceptions/HefestoException.kt
package io.github.robertomike.hefesto.exceptions

/**
 * Base exception class for all Hefesto-related errors.
 * 
 * Thrown when general errors occur during query building or execution.
 */
class HefestoException : RuntimeException {
    /**
     * Creates a new HefestoException with the specified message.
     * @param message the error message
     */
    constructor(message: String) : super(message)

    /**
     * constructor
     * @param message the message
     * @param cause the cause
     */
    constructor(message: String, cause: Throwable) : super(message, cause)
}
